### ISSUE TITLE
Generating template for GCP provider with TypeScript consistently leads to V8 module hitting OOM error

### ISSUE THREAD

**Issue by tylerburdsall** [hashicorp/terraform-cdk#1885]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

* `cdktf` version: `0.11.2`
* `@cdktf/provider-aws`: `8.0.10`
* `@cdktf/provider-google`: `0.9.26`
* `typescript`: `4.4.2`

```
%cat cdktf.json
{
  "language": "typescript",
  "app": "npx ts-node bin/main.ts",
  "projectId": "<removed>",
  "terraformProviders": [],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
```

### Debug Output
https://gist.github.com/tylerburdsall/3b2ffd87ec07e4b5ba43e508ee161601

### Expected Behavior

`cdktf` should have been able to synthesize project without OOM error

### Actual Behavior

The process fails with an OOM error for the Node V8 engine

### Steps to Reproduce

1. Create a project that uses GCP as a provider
2. Build project on an AWS CodeBuild machine (Linux, 15 GB memory, 8 vCPUs)
3. Run `cdktf synth` without adding any additional Node environment variables to increase heap size

### Important Factoids
It looks like this is a known issue that has happened with other providers: https://github.com/hashicorp/terraform-cdk/issues/1264

I can also see that there is a warning in the library itself: https://github.com/hashicorp/terraform-cdk/blob/main/packages/%40cdktf/provider-generator/lib/get/constructs-maker.ts#L320

Is there a reason the `NODE_OPTIONS` flag isn't set across providers (perhaps [here for reference](https://github.com/hashicorp/terraform-cdk/blob/7c823088281f66ec6a70e7297324d8e825ffed29/packages/%40cdktf/provider-generator/lib/get/constructs-maker.ts#L252))?

**Comment by cprice404** [comment]
Another data point on this issue; we were synthing fine with cdktf `v0.10.3`, so this seems like a regression that occurred sometime between those two releases.

**Comment by DanielMSchmidt** [comment]
You can set `NODE_OPTIONS` yourself as a temporary solution. Besides it seems like the amount of code we generate is too much for TS to handle. While there is not much besides improving JSII we can do regarding `cdktf get` performance you could look into changing your app command to be something more performant for the compilation process. 

[esbuild](https://esbuild.github.io/getting-started/) could be a good alternative, in the end all the app command needs to do is to run your program so it can synthesize the configuration. [Deno](https://deno.land/) could be an option as well.

**Comment by ansgarm** [comment]
Hi 👋
We recently released CDKTF 0.13 which puts each resource into their own namespace and due to this, synthesizing should take way less memory as `ts-node` does not need to go over the whole provider at once (if using direct imports with `/lib/` in the import path.
Head over to https://cdk.tf/0.13 to learn more about this recent change and how to adjust the imports.

However, if you still run into performance issues in certain cases, don't hesitate to file a [new issue](https://cdk.tf/bug) and we'll try to reproduce and improve those cases as well 🕵️
